### PR TITLE
Add documentation for the Cadence documentation generator

### PR DESCRIPTION
**Pull request by SupunS**
- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: docs/language/syntax.md [hunk 1]
```diff
@@ -37,6 +37,24 @@ Multi-line comments are balanced.
 /* this is a // comment up to here */ this is not part of the comment */
 ```
 
+### Documentation Comments
+Documentation comments (also known as "doc-strings" ro "doc-comment") are a special set of comments that would be
```
Comment (joshuahannan) [inline docs/language/syntax.md:41]:
```suggestion
Documentation comments (also known as "doc-strings" or "doc-comment") are a special set of comments that would be
```

File: docs/language/syntax.md [hunk 2]
```diff
@@ -37,6 +37,24 @@ Multi-line comments are balanced.
...
+processed by various tools to generate human-readable documentations for cadence programs.
```
Comment (joshuahannan) [inline docs/language/syntax.md:42]:
```suggestion
processed by various tools to generate human-readable documentation for cadence programs.
```

File: docs/language/syntax.md [hunk 3]
```diff
@@ -37,6 +37,24 @@ Multi-line comments are balanced.
...
+
+Single line doc-comments starts with three slashes (`///`)
```
Comment (joshuahannan) [inline docs/language/syntax.md:44]:
```suggestion
Single line doc-comments start with three slashes (`///`)
```

File: tools/docgen/README.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+The documentation comments (i.e: "doc-strings" / "doc-comments": line comments starts with `///`,
+or block comments starting with `/**`) available in Cadence programs are processed by the tool,
+to produce human-readable documentations.
+
+### Markdown Support
+Standard Markdown format is supported in doc-comments, with a bit of Cadence flavour.
+This means, any Markdown syntax used within the comments would be honoured and rendered like a standard Markdown snippet.
+It gives the flexibility for the developers to write well-structured documentations.
+
+e.g: A set of bullet points added using Markdown bullets syntax, would be rendered as bullet points in the
```
Comment (joshuahannan) [inline tools/docgen/README.md:29]:
```suggestion
e.g: A set of bullet points added using Markdown bullets syntax would be rendered as bullet points in the
```

File: docs/language/syntax.md [hunk 5]
```diff
@@ -37,6 +37,24 @@ Multi-line comments are balanced.
 /* this is a // comment up to here */ this is not part of the comment */
 ```
 
+### Documentation Comments
+Documentation comments (also known as "doc-strings" or "doc-comment") are a special set of comments that would be
+processed by various tools to generate human-readable documentation for cadence programs.
```
Comment (turbolent) [inline docs/language/syntax.md:42]:
```suggestion
Documentation comments (also known as "doc-strings" or "doc-comment") are a special set of comments that can be
processed by tools, for example to generate human-readable documentation, or provide documentation in an IDE.
```

File: docs/language/syntax.md [hunk 6]
```diff
@@ -37,6 +37,24 @@ Multi-line comments are balanced.
...
+
+Single line doc-comments start with three slashes (`///`)
```
Comment (turbolent) [inline docs/language/syntax.md:44]:
```suggestion
Doc-comments either start with three slashes (`///`) on each line,
or are surrounded by `/**` and `**/`.

```

File: docs/language/syntax.md [hunk 7]
```diff
@@ -37,6 +37,24 @@ Multi-line comments are balanced.
...
+```cadence
+/// This is a documnetation comment on a single line.
+/// Another documnetation comment line that is not executed.
```
Comment (turbolent) [inline docs/language/syntax.md:47]:
```suggestion
/// This is a documentation comment for `x`.
/// It spans multiple lines.
```

File: docs/language/syntax.md [hunk 8]
```diff
@@ -37,6 +37,24 @@ Multi-line comments are balanced.
...
+
+let x = 1
+```
+
+Multi-line doc-comments comments start with a slash followed by two asterisks (`/**`)
+```cadence
+/** This is a documnetation comment
+ which spans multiple lines. **/
```
Comment (turbolent) [inline docs/language/syntax.md:55]:
```suggestion

```cadence
/** 
  This is a documentation comment
  which also spans multiple lines.
**/
```

File: tools/docgen/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+The tool currently supports generating documentation in Markdown format.
+
+## How To Run
+Navigate to `<cadence_dir>/tools/docgen` directory and run:
+```
+go run main.go <path_to_cadence_file> <output_dir>
+```
+
+## Documentation Comments Format
+The documentation comments (i.e: "doc-strings" / "doc-comments": line comments starts with `///`,
```
Comment (turbolent) [inline tools/docgen/README.md:20]:
```suggestion
The documentation comments ("doc-strings" / "doc-comments": line comments starting with `///`,
```

File: tools/docgen/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+or block comments starting with `/**`) available in Cadence programs are processed by the tool,
+to produce human-readable documentations.
+
+### Markdown Support
+Standard Markdown format is supported in doc-comments, with a bit of Cadence flavour.
+This means, any Markdown syntax used within the comments would be honoured and rendered like a standard Markdown snippet.
```
Comment (turbolent) [inline tools/docgen/README.md:26]: 👍

File: tools/docgen/README.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+Parameters can be documented using the `@param` tag, followed by the parameter name, a colon (`:`) and the parameter description.
+The return type can be documented using the `@return` tag.
+
+```
+/// This is the description of the function. This function adds two values.
+///
+/// @param a: First integer value to add
+/// @param b: Second integer value to add
+/// @return Addition of the two arguments `a` and `b`
+///
```
Comment (turbolent) [inline tools/docgen/README.md:59]: Quite like this mix 👍

File: tools/docgen/README.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+- Use inline-codes (within backticks `` `foo` ``) when referring to names/identifiers in the code.
+  - e.g: Referring to a function name, parameter names, etc.
+    ```
+    /// This is the description of the function.
+    /// This function adds `a` and `b` values.
+    ///
+    pub fun add(a: Int, b: Int): Int {
+    }
+    ```
+- When documenting function parameters and return type, avoid mixing parameter/return-type documentations
```
Comment (turbolent) [inline tools/docgen/README.md:77]: 👍

**Review by turbolent — APPROVED** [review]

**Pull request merged**